snap/squashfs2/directory.go

```diff
@@ -0,0 +1,98 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2014-2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * https://www.kernel.org/doc/html/v5.8/filesystems/squashfs.html
+ */
+
+package squashfs2
+
+import "fmt"
+
+func (d *directory) readDirectoryHeader() squashfs_dir_header {
+	header := make([]byte, 12)
+	d.reader.read(header)
+
+	return squashfs_dir_header{
```

Comment:
(nitpick) Go encourages using mixedCaps (https://go.dev/doc/effective_go#mixed-caps) so it would be more idiomatic to have `dirHeader` here. The prefix `squashf`of `squashfsDirHeader` can probably also be dropped because usually the go module in use gives enough context (if it was exported it would be addressed via `squashfs.DirHeader` and `squashfs.SquashfsDirHeader` would read a bit redundantly).